src/commands/analyze/command.ts

```diff
@@ -256,22 +258,36 @@ export async function analyzeCommand(options: GenezioAnalyzeOptions) {
             );
             debugLogger.debug("Serverless HTTP entry file found:", entryFile);
 
-            // TODO: Add support for detecting and building typescript backends
-            // const isTypescriptFlag = await isTypescript(contents);
+            const isTypescriptFlag = await isTypescript(contents);
+            debugLogger.debug("Is Serverless HTTP typescript:", isTypescriptFlag);
+
+            let entryFileBuildOut = entryFile;
+            if (isTypescriptFlag) {
+                const tsconfigJsonContent = JSON.parse(
+                    await retrieveFileContent(path.join(componentPath, "tsconfig.json")),
+                );
+                entryFileBuildOut = getEntryFileTypescript(entryFile, tsconfigJsonContent);
+                debugLogger.debug("Serverless HTTP entry file build out:", entryFileBuildOut);
+            }
 
             const packageManagerType =
                 genezioConfig.backend?.language?.packageManager || NODE_DEFAULT_PACKAGE_MANAGER;
             const packageManager = packageManagers[packageManagerType];
             await addBackendComponentToConfig(configPath, {
                 path: componentPath,
                 language: {
-                    // TODO: Add support for detecting and building typescript backends
                     name: Language.js,
```

Comment:
name: Language.js should be decided based on isTypescriptFlag

Note: - There are 2 more places, do a find after `name: Language.js` to spot them